storage/innobase/srv/srv0start.cc

```diff
@@ -1988,6 +1985,19 @@ void innodb_preshutdown()
 
   if (srv_read_only_mode)
     return;
+
+  #ifndef EMBEDDED_LIBRARY
+  if (srv_force_recovery < SRV_FORCE_NO_TRX_UNDO && srv_was_started &
+      !trx_rollback_is_active && trx_sys.any_active_transaction_recovered()) {
```

Comment:
It looks like the bitwise arithmetic `&` should be replaced with the logical short-circuit `&&`.
(Also, the `{` needs to be on its own line.)

Reply:
fixed